### ISSUE TITLE
onBlur / onFocus don't bubble in Firefox / macOS

### ISSUE THREAD

**Issue by danburzo** [facebook/react#12993]
Tested with React 16.4 on Firefox 60 / macOS, the `onBlur` and `onFocus` don't bubble to the parent component when focusing / blurring the button:

```jsx
class MyComponent extends React.Component {
  render() {
    return (
      <div 
        onFocus={ e => console.log('focus') } 
        onBlur={ e => console.log('blur') }
      >
        <button>Click me</button>
      </div>
    );
  }
}
```

(@AlmeroSteyn [writes here](https://github.com/reactjs/reactjs.org/pull/923#issuecomment-395319251) that they bubble correctly on Firefox/Windows)

**Comment by AlmeroSteyn** [comment]
Tested this exact example in Firefox 60 on Windows 10 and the events bubble to the parent `div` tag as it does in other browsers tested on Windows.

**Comment by jquense** [comment]
I'm not necessarily seeing that behavior tabbing through the components results in `focus` and `blur` logs as i'd expect

https://codesandbox.io/s/r4k9p6jp4p

**Comment by danburzo** [comment]
@jquense, you're right: the buggy behavior is only with clicking — tabbing through the elements works in FF/Mac. Updated the issue to reflect that.

**Comment by gaearon** [comment]
I'm seeing that in Chrome, the native event target is `button`:

<img width="316" alt="screen shot 2018-08-17 at 6 56 34 pm" src="https://user-images.githubusercontent.com/810438/44281317-50696280-a24f-11e8-917c-5f9d44dbd8bf.png">

But in Firefox, it's document:

<img width="362" alt="screen shot 2018-08-17 at 6 57 07 pm" src="https://user-images.githubusercontent.com/810438/44281337-5fe8ab80-a24f-11e8-949d-3c34ec33d782.png">

We can't determine which component was clicked so that's why you don't receive the event :-/
